pkg/operator/controller/externaldns/pod.go

```diff
@@ -363,6 +364,9 @@ func (b *externalDNSContainerBuilder) fillAzureFields(zone string, container *co
 	// https://github.com/kubernetes-sigs/external-dns/issues/2082
 	container.Args = addTXTPrefixFlag(container.Args)
 
+	// https://github.com/kubernetes-sigs/external-dns/issues/2922
+	container.Args = append(container.Args, fmt.Sprintf("--txt-wildcard-replacement=%s", defaultTXTWildcardReplacement))
```

Comment:
Do you think it would be better to limit this option only for `configv1.AzurePlatformType`? Smaller blast radius? Or do you think it's better to be consistent for all platforms?

Reply:
Whoops, cancel that, I didn't realize the function was specific to the Azure provider already.

Reply:
Yes, this PR aims at fixing the problem which is seen only in Azure DNS as it doesn't allow wildcard records (TXT included) which don't start with the asterisk.